drivers/sensor/icm42688/icm42688_core.c

```diff
@@ -0,0 +1,201 @@
+{
+	icm42688_temp_c((int32_t)raw_val, &val->val1, &val->val2);
+}
+
+static int icm42688_channel_get(const struct device *dev, enum sensor_channel chan,
+				struct sensor_value *val)
+{
+	int res = 0;
+	struct icm42688_sensor_data *data = dev->data;
+
+	icm42688_lock(dev);
+
+	switch (chan) {
+	case SENSOR_CHAN_ACCEL_XYZ:
+		icm42688_convert_accel(&val[0], data->readings[1], &data->dev_data.cfg);
+		icm42688_convert_accel(&val[1], data->readings[2], &data->dev_data.cfg);
+		icm42688_convert_accel(&val[2], data->readings[3], &data->dev_data.cfg);
+		break;
+	case SENSOR_CHAN_ACCEL_X:
+		icm42688_convert_accel(val, data->readings[1], &data->dev_data.cfg);
+		break;
+	case SENSOR_CHAN_ACCEL_Y:
+		icm42688_convert_accel(val, data->readings[2], &data->dev_data.cfg);
+		break;
+	case SENSOR_CHAN_ACCEL_Z:
+		icm42688_convert_accel(val, data->readings[3], &data->dev_data.cfg);
+		break;
+	case SENSOR_CHAN_GYRO_XYZ:
+		icm42688_convert_gyro(&val[0], data->readings[4], &data->dev_data.cfg);
+		icm42688_convert_gyro(&val[1], data->readings[5], &data->dev_data.cfg);
+		icm42688_convert_gyro(&val[2], data->readings[6], &data->dev_data.cfg);
+		break;
+	case SENSOR_CHAN_GYRO_X:
+		icm42688_convert_gyro(val, data->readings[4], &data->dev_data.cfg);
+		break;
+	case SENSOR_CHAN_GYRO_Y:
+		icm42688_convert_gyro(val, data->readings[5], &data->dev_data.cfg);
+		break;
+	case SENSOR_CHAN_GYRO_Z:
+		icm42688_convert_gyro(val, data->readings[6], &data->dev_data.cfg);
+		break;
+	case SENSOR_CHAN_DIE_TEMP:
+		icm42688_convert_temp(val, data->readings[0]);
+		break;
+	default:
+		res = -ENOTSUP;
+		break;
+	}
+
+	icm42688_unlock(dev);
+
+	return res;
+}
+
+static int icm42688_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	uint8_t status;
+	struct icm42688_sensor_data *data = dev->data;
+	const struct icm42688_sensor_config *cfg = dev->config;
+
+	icm42688_lock(dev);
+
+	int res = icm42688_spi_read(&cfg->dev_cfg.spi, REG_INT_STATUS, &status, 1);
+
+	if (res) {
+		goto cleanup;
+	}
+
+	if (!FIELD_GET(BIT_INT_STATUS_DATA_RDY, status)) {
+		res = -EBUSY;
+		goto cleanup;
+	}
+
+	uint8_t readings[14];
+
+	res = icm42688_read_all(dev, readings);
+
+	if (res == 0) {
+		for (int i = 0; i < 7; i++) {
+			data->readings[i] =
+				sys_le16_to_cpu((readings[i * 2] << 8) | readings[i * 2 + 1]);
+		}
+	}
+
+cleanup:
+	icm42688_unlock(dev);
+	return res;
+}
+
+static int icm42688_attr_set(const struct device *dev, enum sensor_channel chan,
+			     enum sensor_attribute attr, const struct sensor_value *val)
+{
+	return -ENOTSUP;
+}
+
+static int icm42688_attr_get(const struct device *dev, enum sensor_channel chan,
+			     enum sensor_attribute attr, struct sensor_value *val)
+{
+	return -ENOTSUP;
+}
+
+static const struct sensor_driver_api icm42688_driver_api = {
+	.sample_fetch = icm42688_sample_fetch,
+	.channel_get = icm42688_channel_get,
+	.attr_set = icm42688_attr_set,
+	.attr_get = icm42688_attr_get,
+};
+
+int icm42688_init(const struct device *dev)
```

Comment:
Nit - I'd prefer that the driver implementation is provided by the module matching the sensor name.  So this file would be `icm42688.c` and the file with the helper routines `icm42688_reset()` and `icm42688_configure()` would be called `icm42688_core.c` or maybe `icm42688_support.c`.

@MaureenHelm what your thoughts?

Reply:
I ended with `icm42688_common.c` and `icm42688.c`, does that work?